tests/unit/test_path_template.py

```diff
@@ -362,6 +362,7 @@ def test_transcode_with_additional_bindings(
         [[["get", "/v1/{name}", ""]], {"name": "first/last"}],
         [[["get", "/v1/{name=mr/*/*}", ""]], {"name": "first/last"}],
         [[["post", "/v1/{name}", "data"]], {"name": "first/last"}],
+        [[["post", "/v1/{first_name}", "data"]], {"last_name": "last"}],
     ],
 )
 def test_transcode_fails(http_options, request_kwargs):
```

Comment:
Could you have this test check the exception string?

https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions